proxmox/api/credentials.go

```diff
@@ -16,41 +17,60 @@ const rootUsername = "root@pam"
 // Credentials is a struct that holds the credentials for the Proxmox Virtual
 // Environment API.
 type Credentials struct {
-	Username string
-	Password string
-	OTP      *string
-	APIToken *string
+	AuthTicket          string
+	CSRFPreventionToken string
+	APIToken            *string
+	OTP                 *string
+	Username            string
+	Password            string
 }
 
 // NewCredentials creates a new Credentials struct.
-func NewCredentials(username, password, otp, apiToken string) (*Credentials, error) {
-	if apiToken != "" {
+//
+//nolint:lll
+func NewCredentials(username, password, otp, apiToken string, authTicket string, csrfPreventionToken string) (*Credentials, error) {
+	errorAuthTicketCommonMsg := "the Proxmox Virtual Environment API requires auth params; "
+
+	switch {
+	case authTicket != "" && csrfPreventionToken != "":
+		switch {
+		case authTicket == "":
+			return nil, errors.New(errorAuthTicketCommonMsg + "detected csrfPreventionToken, but authTicket is unset")
+		case csrfPreventionToken == "":
+			return nil, errors.New(errorAuthTicketCommonMsg + "detected authTicket, but csrfPreventionToken is unset")
+		}
```

Comment:
Conditions in the inner `switch` will never be true...

But I'm thinking about refactoring the whole credentials thing.